src/main/scala-2.11/org/apache/james/gatling/jmap/JmapMessages.scala

```diff
@@ -42,6 +44,41 @@ object JmapMessages {
           "#0"
           ]]"""))
   
+  def retrieveMessageIds(mailboxId: String) = 
+    JmapAuthentication.authenticatedQuery("retrieveMessageIds", "/jmap")
+      .body(StringBody(
+        s"""[[
+          "getMessageList",
+          {
+            "filter": {
+              "inMailboxes" : [ "$mailboxId" ]
+            }
+          },
+          "#0"
+          ]]"""))
+      .check(saveMessageIds)    
+
+  def saveMessageIds = {
+    jsonPath(messageIdsPath).findAll.saveAs("messageIds")
+  }
+
+  def moveMessagesToMailboxId =
+    foreach("${messageIds}", "messageId") {
```

Comment:
something like that (did not compile, it's just pseudocode) : 
```
exec((s: Session) => s.setAttribute("update", 
  s("messageIds").map(x => """
  "$${x}" : {
                  "mailboxIds": [ "$${mailboxId}" ]"""
).join(","))
.exec(JmapAuthentication.authenticatedQuery("moveSentMessagesToMessageId", "/jmap")
        .body(StringBody(
          s"""[[
            "setMessages",
            {
              "update": {
                "$${update}
                }
              }
            },
            "#0"
            ]]""")))
```

Reply:
Thx